Source/Core/Core/Lua/LuaFunctions/LuaEmuFunctions.cpp

```diff
@@ -0,0 +1,137 @@
+#include "LuaEmuFunctions.h"
+#include "../LuaVersionResolver.h"
+#include "Core/State.h"
+#include "Core/Core.h"
+#include "Core/Movie.h"
+#include "Core/PowerPC/PowerPC.h"
+#include <optional>
+
+namespace Lua
+{
+namespace LuaEmu
+{
+  bool waitingForSaveStateLoad = false;
+  bool waitingForSaveStateSave = false;
+  bool waitingToStartPlayingMovie = false;
+  bool waitingToSaveMovie = false;
+
+  std::string loadStateName;
+  std::string saveStateName;
+  std::string moviePathName;
+  std::string playMovieName;
+  std::optional<std::string> blankString;
+  std::string saveMovieName;
+
+  class emu
+{
+public:
+  inline emu() {}
+};
+
+emu* emuPointer = NULL;
```

Comment:
These seem like they should be `unique_ptr` instances, so that the memory is always managed, since they don't seem to be deleted on deinitialization, which will add a lot of false-positives to memory sanitizing tools and leak checkers.